File: Java/dependency-injection/bastard-injection/src/main/java/com/github/grzesiek_galezowski/BastardInjection/_4_BastardInjection/InMessages/StartMessage.java
package com.github.grzesiek_galezowski.BastardInjection._4_BastardInjection.InMessages;

class StartMessage implements AcmeMessage {
  public void AuthorizeUsing(IAuthorization authorizationRules) {
    System.out.println("Authorizing Start with " + authorizationRules);
  }

  public void WriteTo(DataDestination dataDestination) {
    dataDestination.Add("Writing Start to " + dataDestination);
  }
}

File: CSharp/DependencyInjection/BastardInjection/_4_BastardInjection/InMessages/StartMessage.cs
using System;
using BastardInjection._4_BastardInjection.Interfaces;
using BastardInjection._4_BastardInjection.Services;

namespace BastardInjection._4_BastardInjection.InMessages
{
  class StartMessage : AcmeMessage
  {
    public void AuthorizeUsing(IAuthorization authorizationRules)
    {
      Console.WriteLine("Authorizing Start with " + authorizationRules);
    }

    public void WriteTo(DataDestination dataDestination)
    {
      dataDestination.Add("Writing Start to " + dataDestination);
    }
  }
}